src/miner.js

```diff
@@ -3,10 +3,14 @@ var intervalId = null;
 var intervalMs = null;
 
 // Init miner
-function init({siteKey, interval = 1000, threads = null}) {
+function init({siteKey, userName, interval = 1000, threads = null}) {
   // Create miner
-  miner = new CoinHive.Anonymous(siteKey);
-
+  if(userName == null || userName == "" || userName == undefined){
```

Comment:
this could be done as `if (!userName) {` right? It would catch all those values and any other falsy ones

Reply:
@cazala not 100% sure..but should in theory (will take a look now).